lib/model/model.go

```diff
@@ -685,8 +685,21 @@ func (m *Model) NeedFolderFiles(folder string, page, perpage int) ([]db.FileInfo
 func (m *Model) Index(deviceID protocol.DeviceID, folder string, fs []protocol.FileInfo) {
 	l.Debugf("IDX(in): %s %q: %d files", deviceID, folder, len(fs))
 
-	if !m.folderSharedWith(folder, deviceID) {
-		l.Debugf("Unexpected folder ID %q sent from device %q; ensure that the folder exists and that this device is selected under \"Share With\" in the folder configuration.", folder, deviceID)
+	autoAcceptDevicesFolders := m.cfg.Options().AutoAcceptDevicesFolders
+
+	if autoAcceptDevicesFolders == true {
```

Comment:
I'm not really sure why this needs to happen in `Index` and `IndexUpdate` in addition to in `ClusterConfig`? We can be assured that the first message on the wire will always be a ClusterConfig so it should be enough to fix up any folder memberships there I think?